devicetypes/smartthings/zigbee-window-shade.src/zigbee-window-shade.groovy

```diff
@@ -77,19 +77,20 @@ def parse(String description) {
     if (description?.startsWith("read attr -")) {
         Map descMap = zigbee.parseDescriptionAsMap(description)
         if (descMap?.clusterInt == CLUSTER_WINDOW_COVERING && descMap.value) {
-            log.debug "attr: ${descMap?.attrInt}, value: ${descMap?.value}, decValue: ${Integer.parseInt(descMap.value, 16)}, ${device.getDataValue("model")}"
+            log.debug "attr: ${descMap?.attrInt}, value: ${descMap?.value}, descValue: ${Integer.parseInt(descMap.value, 16)}, ${device.getDataValue("model")}"
             List<Map> descMaps = collectAttributes(descMap)
             def liftmap = descMaps.find { it.attrInt == ATTRIBUTE_POSITION_LIFT }
             if (liftmap) {
-                 if (liftmap.value == "64") { //open
+                state.level = Integer.parseInt(liftmap.value, 16)
+                if (liftmap.value == "64") { //open
                     sendEvent(name: "windowShade", value: "open")
                     sendEvent(name: "level", value: "100")
                 } else if (liftmap.value == "00") { //closed
                     sendEvent(name: "windowShade", value: "closed")
                     sendEvent(name: "level", value: "0")
                 } else {
                     sendEvent(name: "windowShade", value: "partially open")
-                    sendEvent(name: "level", value: zigbee.convertHexToInt(attr.value))
+                    sendEvent(name: "level", value: zigbee.convertHexToInt(liftmap.value))
```

Comment:
Coming to this too late, but if we get a chance in the future, I'd like to convert all of these string comparisons to int comparisons, since you already did the conversion and stored it in `state.level`. This second conversion down here also didn't need to be done.